docs/content/core/volumes.md

```diff
@@ -5,228 +5,176 @@ aliases:
+Instead Docksal defines several named Docker volumes per project for you:
+ 
+- `project_root` stores your project files (see warnings below abut usecases of this volume)
+- `cli_home` to store home folder of your `cli` container independently of the `cli` container
+- `db_data` to store your database data independently of the `db` container 
+- `docksal_ssh_agent` to share SSH keys with containers
 
-Instead of using a host path every time we want to mount a volume, we can give the volume a name and refer to it by name: 
+Docksal made these volumes to function transparently across different operating systems, but their actual definitions 
+vary depending on the `DOCKSAL_VOLUMES` option.
 
-```yaml
-version: "2.1"
+{{% notice note %}}
+`project_root` volume is mapped to a specific location on the host. 
+If you need to move the location of a project on the computer, remove the project stack first with
+`fin project rm`, then you can move the project folder and start the project stack again.
+{{% /notice %}}
 
-services:
-  cli:
-    volumes:
-      # Project root volume
-      - project_root:/var/www:rw,nocopy,cached
-      # Shared ssh-agent socket
-      - docksal_ssh_agent:/.ssh-agent:ro
-...
-
-volumes:
-  project_root:
-    driver: local
-    driver_opts:
-      type: none
-      device: ${PROJECT_ROOT}
-      o: bind
-  docksal_ssh_agent:
-    external: true
-```
+{{% notice warning %}}
+When customizing `docksal.yml` make sure to include the `cached` option **anywhere** the `project_root` volume is attached
+to a service. Mixing `cached` and non-`cached` mounts for the same volume in your project stack will lead to 
+issues and errors with Docker Desktop on macOS. See [docksal/docksal#678](https://github.com/docksal/docksal/issues/678) for more details.
+{{% /notice %}}
 
-In the example above, `project_root` and `docksal_ssh_agent` are "named volumes." The first one is a project level one,
-while the second one is a global volume and is used by all projects.
+## DOCKSAL_VOLUMES
 
-See [stacks/volumes-bind.yml](https://github.com/docksal/docksal/blob/master/stacks/volumes-bind.yml).
+`DOCKSAL_VOLUMES` value changes the driver option for project volumes mentioned above, and also affects some additional `fin` behavior. 
 
-Defining volumes this way makes it much easier to override volume settings in one place (`volumes` section) vs multiple 
-places in the yaml file. We can now swap bind mounting with something else. See below.
+### bind 
 
-### osxfs:cached Mode with Docker for Mac
+With this option containers access files using **Docker bind** driver option, which basically means direct access. 
 
-As explained above for bind volumes option the files are actually mapped to the VM that Docker runs in
-via `osxfs:cached`. Docksal automatically enables the `osxfs:cached` mode on Docker for Mac.
+While Docker can access files directly on Linux, on macOS and Windows it works inside VM (VirtualBox or xhyve with Docker Desktop), 
```

Comment:
it's not a specific VM in this case

Reply:
It is specific as far as it is the VM operating. “A” VM (for something generic) doesn’t work because you are specifying an operation. It’s just a quirky American English thing that requires a definite article.